src/Compilers/VisualBasic/Portable/Symbols/ConstraintsHelper.vb

```diff
@@ -1110,28 +1094,61 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
             Return New CompoundDiagnosticInfo(diagnostics)
         End Function
 
+        Public Enum ConstructorConstraintError
+            None
+            NoPublicParameterlessConstructor
+            MustInheritType
+            HasRequiredMembers
+        End Enum
+
         ''' <summary>
         ''' Return true if the class type has a public parameterless constructor.
         ''' </summary>
```

Comment:
>If TypeOf constructor.ContainingType Is SourceMemberContainerTypeSymbol Then [](http://example.com/codeflow?start=24&length=76)

I think this check won't work for retargeting scenarios. I do not think it is worth adding extra complexity with the goal to suppress extra errors. If we really think it is important to do, we should consider changing implementation of ```AllRequiredMembers``` to return an empty set instead. And, perhaps, adjust behavior of ```HasRequiredMembersError``` as well.

Reply:
It looks like codeflow messed this comment up, I'm not sure what it was supposed to be on, but I see it on the Function header of `HasPublicParameterlessConstructor`

Reply:
> It looks like codeflow messed this comment up, I'm not sure what it was supposed to be on ...

I think the comment was about ```If TypeOf constructor.ContainingType Is SourceMemberContainerTypeSymbol``` added below.

Reply:
The code in question has been removed.